DeBERTa/DeBERTa-v2/SEW Support for torch 1.11

LysandreJik opened the pull request:
The internal `torch` method `_softmax_backward_data` changed API between 1.10 and 1.11, from requiring a tensor as its last output to requiring a size.

This PR updates the concerned models so that they are correctly supported.

Torch 1.11: https://github.com/pytorch/pytorch/blame/e47a5a64bbf4d388b70397e3237f9d5710ee4c9c/tools/autograd/derivatives.yaml#L1861
Before: https://github.com/pytorch/pytorch/blame/768cfaa8f86bf7c7b0af441d1536f060274c27a0/tools/autograd/derivatives.yaml#L1704

LysandreJik committed:
Support for torch 1.11

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_16043). All of your documentation changes will be reflected on that endpoint.

LysandreJik commented on src/transformers/models/deberta/modeling_deberta.py:
```diff
@@ -37,6 +38,8 @@
 
 logger = logging.get_logger(__name__)
 
+convert_to_dtype = not version.parse(torch.__version__) < version.parse("1.11")
```
Maybe a better name for that value would be something more specific to its purpose, but then the name starts being long and @sgugger gets angry, wdyt?

```suggestion
convert_softmax_tensor_to_dtype = not version.parse(torch.__version__) < version.parse("1.11")
```

patrickvonplaten replied:
I'd maybe just add a `do_convert...` to make it a bit clearer it's a flag and not a function

patrickvonplaten replied:
don't mind the name after :-)

patrickvonplaten reviewed (approved):
Thanks!

sgugger reviewed (commented):
For the torch int div we wrote our own function that does the test internally. I think we should do this the same way and write our own `_softmax_backward_data` inside `pytorch_utils` which will do the test internally, then import this one here.

LysandreJik committed:
Address Sylvain's comment

LysandreJik commented:
Addressed your comment @sgugger, could you do a second review? 

As seen with Sylvain offline, I've moved out the `packaging.version.parse` operation of the methods, as otherwise they would be called inside the methods themselves, which are called multipled times in forward passes. @patrickvonplaten could you check if that's fine with you?

LysandreJik commented on src/transformers/pytorch_utils.py:
```diff
@@ -14,18 +14,34 @@
 
 logger = logging.get_logger(__name__)
 
+is_torch_less_than_1_8 = version.parse(torch.__version__) < version.parse("1.8.0")
+is_torch_less_than_1_11 = version.parse(torch.__version__) < version.parse("1.11")
```
The torch version cannot change during runtime, so this is harmless

sgugger commented:
I don't think we have a patch planned. We will have 4.18 released probably next week instead :-)

speediedan commented:
> I don't think we have a patch planned. We will have 4.18 released probably next week instead :-)

AWESOME! looking forward to it!

whoknowsB commented:
Note that this evaluates True for pre-releases such as '1.11.0a0+b6df043'. So the the error is still present.
`is_torch_less_than_1_11 = version.parse(version.parse(torch.__version__).base_version) < version.parse("1.11")` may help.
Am I missing something?
